source/common/upstream/load_balancer_impl.cc

```diff
@@ -774,6 +780,10 @@ void EdfLoadBalancerBase::refresh(uint32_t priority) {
 }
 
 HostConstSharedPtr EdfLoadBalancerBase::peekAnotherHost(LoadBalancerContext* context) {
+  if (stashed_random_.size() + 1 > total_healthy_hosts_) {
+    return nullptr;
+  }
```

Comment:
This one isn't a ratio thing - we currently cap #prefetches to the number of healthy hosts.

Reply:
Are there plans to relax that restriction?  It seems to get in the way of getting to a fixed number of connections in cases where the number of healthy upstreams is less than the desired number of connections.

Reply:
as needed.  Right now I was aiming at the server side, where for low QPS you'd genreally prefetch a few, and for high qps you'd want per upstream but as we move towards mobile if we only have a single upstream per endpoint we'd want to have more than 1 prefetch.
I don't want it unlimited because it's wasteful and it acts as a useful upper bound.